src/inflators/spawner.ts

```diff
@@ -0,0 +1,44 @@
+import { addComponent } from "bitecs";
+import { HubsWorld } from "../app";
+import {
+  CursorRaycastable,
+  HandCollisionTarget,
+  ObjectSpawner,
+  RemoteHoverTarget,
+  SingleActionButton
+} from "../bit-components";
+import { OBJECT_SPAWNER_FLAGS } from "../bit-systems/object-spawner";
+import { inflateMediaLoader } from "./media-loader";
+
+export interface SpawnerParams {
+  src: string;
+  mediaOptions?: {
+    applyGravity?: boolean;
+  };
+}
+
+export function inflateSpawner(world: HubsWorld, eid: number, props: SpawnerParams) {
+  inflateMediaLoader(world, eid, {
+    src: props.src,
+    recenter: false,
+    resize: false,
+    animateLoad: false,
+    isObjectMenuTarget: false
+  });
+
+  addComponent(world, HandCollisionTarget, eid);
+  addComponent(world, CursorRaycastable, eid);
+  addComponent(world, RemoteHoverTarget, eid);
+  addComponent(world, SingleActionButton, eid);
+
+  addComponent(world, ObjectSpawner, eid);
+  ObjectSpawner.src[eid] = APP.getSid(props.src);
+  ObjectSpawner.flags[eid] = props.mediaOptions?.applyGravity ? OBJECT_SPAWNER_FLAGS.APPLY_GRAVITY : 0;
+
+  // TODO rigidbody+shape for hand grabbing
```

Comment:
Ah perhaps. It should hopefully be much more straightforward though. We also may be able to avoid that whole class of problem if we rethink physics shapes a bit.